packages/api/tests/models/rules/test-rules-model.js

```diff
@@ -328,6 +334,137 @@ test.serial('Updaing a kinesis type rule value results in new event source mappi
   await deleteKinesisEventSourceMappings();
 });
 
+test.serial('updateRuleTrigger() a kinesis type rule value does not delete existing source mappings', async (t) => {
+  const { kinesisRule } = t.context;
+
+  // create rule trigger and rule
+  const kinesisArn1 = `arn:aws:kinesis:us-east-1:000000000000:${randomId('kinesis1')}`;
+  kinesisRule.rule.value = kinesisArn1;
+  await rulesModel.create(kinesisRule);
+
+  const rule = await rulesModel.get({ name: kinesisRule.name });
+  t.teardown(async () => {
+    await rulesModel.delete(rule);
+    await deleteKinesisEventSourceMappings();
+  });
+
+  // update rule value
+  const updates = {
+    name: rule.name,
+    rule: { ...rule.rule, value: `arn:aws:kinesis:us-east-1:000000000000:${randomId('kinesis2')}` },
+  };
+
+  const ruleWithUpdatedTrigger = await rulesModel.updateRuleTrigger(rule, updates);
+  const updatedRule = await rulesModel.update(ruleWithUpdatedTrigger);
+
+  t.is(updatedRule.name, rule.name);
+  t.not(updatedRule.rule.value, rule.rule.value);
+
+  // Event source mappings exist and have been updated
+  t.truthy(updatedRule.rule.arn);
+  t.not(updatedRule.rule.arn, rule.rule.arn);
+  t.truthy(updatedRule.rule.logEventArn);
+  t.not(updatedRule.rule.logEventArn, rule.rule.logEventArn);
+
+  const kinesisEventMappings = await getKinesisEventMappings();
+  const consumerEventMappings = kinesisEventMappings[0].EventSourceMappings;
+  const logEventMappings = kinesisEventMappings[1].EventSourceMappings;
+
+  t.is(consumerEventMappings.length, 2);
+  t.is(consumerEventMappings.filter((mapping) => mapping.EventSourceArn === kinesisArn1).length, 1);
+  t.is(logEventMappings.length, 2);
+  t.is(logEventMappings.filter((mapping) => mapping.EventSourceArn === kinesisArn1).length, 1);
+});
+
+test.serial('updateRuleTrigger() a SNS type rule value does not delete existing source mappings', async (t) => {
+  const topic1 = await awsServices.sns().createTopic({ Name: randomId('topic1_') }).promise();
+  const topic2 = await awsServices.sns().createTopic({ Name: randomId('topic2_') }).promise();
+
+  // create rule trigger and rule
+  const snsRule = fakeRuleFactoryV2({
+    workflow,
+    rule: {
+      type: 'sns',
+      value: topic1.TopicArn,
+    },
+    state: 'ENABLED',
+  });
+
+  await rulesModel.create(snsRule);
+
+  const rule = await rulesModel.get({ name: snsRule.name });
+  t.teardown(() => rulesModel.delete(rule));
+
+  // update rule value
+  const updates = {
+    name: rule.name,
+    rule: { ...rule.rule, value: topic2.TopicArn },
+  };
+
+  const ruleWithUpdatedTrigger = await rulesModel.updateRuleTrigger(rule, updates);
+  const updatedRule = await rulesModel.update(ruleWithUpdatedTrigger);
+
+  t.is(updatedRule.name, rule.name);
+  t.not(updatedRule.rule.value, rule.rule.value);
+
+  // Event source mappings exist and have been updated
+  t.truthy(updatedRule.rule.arn);
+  t.not(updatedRule.rule.arn, rule.rule.arn);
```

Comment:
Should we delete the SNS topics in the teardown of this test?

Reply:
3b94347f96041e44edeff31bb8420a4a54582af0